expression/builtin_string.go

```diff
@@ -2916,45 +2916,53 @@ func (b *builtinOrdSig) Clone() builtinFunc {
 // See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_ord
 func (b *builtinOrdSig) evalInt(row chunk.Row) (int64, bool, error) {
 	str, isNull, err := b.args[0].EvalString(b.ctx, row)
+	strBytes := hack.Slice(str)
 	if isNull || err != nil {
 		return 0, isNull, err
 	}
 
-	ord, err := chooseOrdFunc(b.args[0].GetType().Charset)
+	charSet := b.args[0].GetType().Charset
+	ord, err := chooseOrdFunc(charSet)
 	if err != nil {
 		return 0, false, err
 	}
-	return ord(str), false, nil
+
+	enc := charset.NewEncoding(charSet)
+	leftMost, err := enc.EncodeFirstChar(nil, strBytes)
+	if err != nil {
+		return 0, false, err
+	}
+	return ord(leftMost), false, nil
 }
 
-func chooseOrdFunc(charSet string) (func(string) int64, error) {
+func chooseOrdFunc(charSet string) (func([]byte) int64, error) {
 	// use utf8 by default
 	if charSet == "" {
 		charSet = charset.CharsetUTF8
 	}
 	desc, err := charset.GetCharsetInfo(charSet)
 	if err != nil {
-		return nil, err
+		// TODO: After gbk added to charset, this compare should be removed
```

Comment:
Could we remove this and put it to `TestOrd`(comment out related to the test )? I think we can pass tests in `new_character_set_builtin.test`.

Reply:
Resolved.